source/Coop.Core/Server/Services/Sync/Handlers/PeerQueueOverloadedHandler.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Server.Services.Time.Messages;
+using GameInterface.Services.GameDebug.Messages;
+using GameInterface.Services.Heroes.Enum;
+using GameInterface.Services.Heroes.Messages;
+using LiteNetLib;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Coop.Core.Server.Services.Sync.Handlers
+{
+    internal class PeerQueueOverloadedHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+
+        private readonly Poller poller;
+        private readonly HashSet<NetPeer> overloadedPeers;
+
+        private TimeControlEnum originalSpeed;
+
+        private readonly TimeHandler timeHandler;
+
+        public PeerQueueOverloadedHandler(
+            IMessageBroker messageBroker, 
+            INetwork network,
+            TimeHandler timeHandler
+        )
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            this.timeHandler = timeHandler;
+
+            overloadedPeers = new();
+            poller = new(Poll, TimeSpan.FromMilliseconds(100));
+
+            messageBroker.Subscribe<PeerQueueOverloaded>(Handle);
+
+            poller.Start();
+        }
+
+        private void Handle(MessagePayload<PeerQueueOverloaded> payload) 
+        {
+            Monitor.Enter(overloadedPeers);
```

Comment:
I dislike using monitor here, use lock instead

Reply:
Using lock locks the resource for the whole duration of the operation, which i don't require. It'll just slow it down.

Reply:
Nevermind, i see your point. I'll use locks instead.